### PR TITLE
CrowdStrike telemetry integration

### PR DESCRIPTION
**Pull request by vg-svitla**
1. CrowdStrike telemetry integration
2. Fix in test_formats.py to skipp checking `parsing_duration_ms`

### CODE REVIEW

File: CrowdStrike/crowdstrike-telemetry/tests/telemetry_event_7.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  "expected": {
+    "message": "{\"event_simpleName\": \"FirewallDeleteRule\", \"ContextTimeStamp\": \"1687509723.415\", \"UserName\": \"j.doe\", \"ConfigStateHash\": \"2181989539\", \"ContextProcessId\": \"1653642597100\", \"InterfaceVersion\": \"65536\", \"aip\": \"4.3.2.1\", \"RpcClientThreadId\": \"45955897880875\", \"FirewallRuleId\": \"{CE605B68-BD8A-4D72-BA66-15341408C483}\", \"AuthenticationId\": \"1077281\", \"ConfigBuild\": \"1007.3.0016810.10\", \"event_platform\": \"Win\", \"TokenType\": \"1\", \"RpcOpNum\": \"7\", \"Entitlements\": \"15\", \"name\": \"FirewallDeleteRuleV3\", \"InterfaceGuid\": \"52C47ED7-2CF7-4670-8EE1-49F729A804C8\", \"RpcClientProcessId\": \"1655394096661\", \"id\": \"a83b4482-ca45-4d6b-850b-79d8a1b9e60f\", \"EffectiveTransmissionClass\": \"3\", \"aid\": \"111111111111111\", \"RpcNestingLevel\": \"0\", \"timestamp\": \"1687509723940\", \"cid\": \"222222222222222222222\"}",
```
Comment (squioc) [inline CrowdStrike/crowdstrike-telemetry/tests/telemetry_event_7.json:12]: WE can extract `UserName` into `user.name`

File: CrowdStrike/crowdstrike-telemetry/ingest/parser.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        output_field: datetime
+    filter: "{{parsed_event.message.ProcessEndTime != None}}"
+
+  - name: set_common_fields
+
+stages:
+  set_common_fields:
+    actions:
+      - set:
+          "@timestamp": "{{event_date.datetime}}"
```
Comment (squioc) [inline CrowdStrike/crowdstrike-telemetry/ingest/parser.yml:38]:
```suggestion
          "@timestamp": "{{event_date.datetime}}"
          "event.action": "{{parsed_event.message.event_simpleName}}"
```

**Review by squioc — CHANGES_REQUESTED** [review]

File: CrowdStrike/crowdstrike-telemetry/ingest/parser.yml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  - name: set_registry_fields
+
+stages:
+  set_registry_fields:
+    actions:
+      - set:
+          "registry.value": "{{parsed_event.message.RegValueName}}"
+          "registry.key": "{{parsed_event.message.RegObjectName}}"
+          "registry.path": "{{parsed_event.message.RegObjectName}}\\{{parsed_event.message.RegValueName}}"
+          "registry.hive": "MACHINE"
```
[the remark this answers is not in the document]
Comment (vg-svitla) [reply CrowdStrike/crowdstrike-telemetry/ingest/parser.yml]: Fixed thank you!

File: CrowdStrike/crowdstrike-telemetry/ingest/parser.yml [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  - name: set_common_fields
+  - name: set_dns_fields
+  - name: set_registry_fields
+
+stages:
+  set_registry_fields:
+    actions:
+      - set:
+          "registry.value": "{{parsed_event.message.RegValueName}}"
+          "registry.key": '{{parsed_event.message.RegObjectName.split("\\")[2]}}'
```
Comment (squioc) [inline CrowdStrike/crowdstrike-telemetry/ingest/parser.yml:41]: Duplicated with L45.
```suggestion
```

File: CrowdStrike/crowdstrike-telemetry/ingest/parser.yml [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+      - set:
+          "registry.key": '{{"\\".join(parsed_event.message.RegObjectName.split("\\")[3:])}}'
+          "registry.hive": '{{parsed_event.message.RegObjectName.split("\\")[2]}}'
+        filter: '{{parsed_event.message.get("RegObjectName") != None and "\\" in parsed_event.message.get("RegObjectName")}}'
+
+      - set:
+          "registry.path": "{{parsed_event.message.RegObjectName}}\\{{parsed_event.message.RegValueName}}"
```
Comment (squioc) [inline CrowdStrike/crowdstrike-telemetry/ingest/parser.yml:49]:
```suggestion
          "registry.path": "{{"\\".join(parsed_event.message.RegObjectName.split("\\")[2:])}}\\{{parsed_event.message.RegValueName}}"
```

**Review by squioc — APPROVED** [review]
:+1:

**Pull request merged**